package.json

```diff
@@ -10,6 +10,9 @@
     "style"
   ],
   "type": "module",
+  "dependencies": {
+    "yaml": "^2.3.1"
```

Comment:
This is a big dependency, and potentially controversial - on an old team we found js-yaml more performant and smaller.

I think it should be left as an "exercise to the reader" to parse yaml into a json object. That way end users could install their preferred yaml library before passing a plain old JavaScript object to crosswords-js. 

No objections to it being a dev dependency for the sake of the demo site.

Reply:
Hi Misha!

> This is a big dependency, and potentially controversial - on an old team we found js-yaml more performant and smaller.
> 
I've had a quick look at [js-yaml](https://www.npmjs.com/package/js-yaml) - great tip - thanks!

> I think it should be left as an "exercise to the reader" to parse yaml into a json object. That way end users could install their preferred yaml library before passing a plain old JavaScript object to crosswords-js.
> 
> No objections to it being a dev dependency for the sake of the demo site.

I understand your concern.  Our points of differentiation to date are no dependencies, a small footprint, and universal usage (pure JS).

YAML makes manual document editing feasible. Looking at the JSON in [**ipuz**](http://justsolve.archiveteam.org/wiki/IPUZ#Sample_data), for example - there's little chance people would regularly construct a 15x15 puzzle by hand! 

Simplicity is the third of @dwmkerr Dave's design goals. I think the puzzle format plays into this. I would like to support YAML for the sake of puzzle makers. Perhaps not in the core repo? I've spoken to @dwmkerr about managing framework samples as separate repos, perhaps another repo for useful utilities, like YAML imports?

Once we move beyond one repo, we should move to an organization account on GitHub, as is the pattern. GitHub handles link forwarding, so we wouldn't lose anyone in the transition.

Reply:
Here's how I see the best of all worlds being achieved:

- keep all dependencies including yaml out of `crosswords-js` - which could become the core crosswords _renderer_ package. I am using it (and a big fan!) on a site I'm building now. I don't need yaml because the main use case is crossword _playing_, not crossword _setting_. The setters are uploading .puz files which I'm parsing using another library. Yaml would be baggage in this package in my opinion
- add yaml and maybe some other opinionated dependencies in a package called something like `@crosswords-js/set` which is designed to help people write new crosswords - this should be separate because it's a whole different set of concerns, I think
- if there are more spin-off packages for slightly different use cases, those could also go under the (for now non existent) `@crosswords-js/` npm scope. 

That may sound like scope creep for this project but I would say if that's the case then it'd be better to keep yaml out for now. When we switched to js-yaml on my old team we still had problems keeping it out of our client bundle. 

The above idea could either be implemented with another repo like @pvspain suggested or by converting this into a multi-package monorepo - monorepo tooling these days is very good for npm packages. My recommendation would be to do that and use turbo.build. I'd be happy to help set this up if interested.